Integration test for ProxyStatus and Experimental AuthZ Check

esnible opened the pull request:
Part one of the fix for https://github.com/istio/istio/issues/23042

Once the istioctl subcommands that query Envoy are tested during integration we will remove the static unit tests for those subcommands.

cc @howardjohn

esnible committed:
Integration test for ProxyStatus and Experimental AuthZ Check

howardjohn commented on tests/integration/pilot/istioctl_test.go:
```diff
@@ -293,3 +293,79 @@ func jsonUnmarshallOrFail(t *testing.T, context, s string) interface{} {
+			// the printing code printed it.
+			g.Expect(output).To(gomega.ContainSubstring(fmt.Sprintf("%s.%s", podID, ns.Name())))
+
+			args = []string{"--namespace=dummy",
```
why do we need dummy namespace?

howardjohn commented on tests/integration/pilot/istioctl_test.go:
```diff
@@ -293,3 +293,79 @@ func jsonUnmarshallOrFail(t *testing.T, context, s string) interface{} {
+			output, _ = istioCtl.InvokeOrFail(t, args)
+			// Just verify pod A is known to Pilot; implicitly this verifies that
+			// the printing code printed it.
+			g.Expect(output).To(gomega.ContainSubstring("0.0.0.0_80"))
```
This doesn't really test much though.. The code could be completely wrong and not catch this.

howardjohn commented on tests/integration/pilot/istioctl_test.go:
```diff
@@ -293,3 +293,79 @@ func jsonUnmarshallOrFail(t *testing.T, context, s string) interface{} {
+				Inject: true,
+			})
+
+			var a echo.Instance
```
why not have a single pod shared between the tests then just run a serious of `istioctl` commands against it?

esnible replied:
I'd like to do that.  Is there a way that I can have multiple TestXYX() functions run after the same pod is ready?  Or is it expected that I have a single Test() function and call out to functions for each subcommand?

howardjohn replied:
Here is how you could do it (pseudocode, may be slightly off)

```
func TestIstioctl() {
  framework.NewTest.Run(func() {
    ns := ...
   echoInstance := setupInstance
   ctx.NewSubtest("describe").Run(func() { ... run istioctl test ...})
   ctx.NewSubtest("authz").Run(func() { ... run istioctl test ...})
}
```

esnible replied:
I believe these tests already run in parallel.   I like the look of that and will use that style in a follow-up.

howardjohn replied:
You can chose parallel or serial with Run or RunParallel I think. Parallel seems reasonable here

howardjohn reviewed (commented):
overall looks good

esnible committed:
Verify 'x authz check' includes the name of the v1beta1 AuthorizationPolicy applied

istio-testing commented:
@esnible: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-local-tests_istio | b8a1c42d37ae110c85bf782060a0730f19f0c3da | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/integ-local-tests_istio/3970) | `/test integ-local-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

howardjohn commented on tests/integration/pilot/istioctl_test.go:
```diff
@@ -293,3 +293,80 @@ func jsonUnmarshallOrFail(t *testing.T, context, s string) interface{} {
+			output, _ = istioCtl.InvokeOrFail(t, args)
+			// Verify the output includes a policy "integ-test", which is the policy
+			// loaded above from authz-a.yaml
+			g.Expect(output).To(gomega.MatchRegexp("noneSDS: default.*\\[integ-test\\]"))
```
Not in scope for this PR, but this is exposing a bug. I am pretty sure its not supposed to say `noneSDS: default`

howardjohn reviewed (approved):
LGTM, can we merge these as a followup and probably also file a bug for authz output?

esnible commented:
/retest

esnible commented:
Can anyone from @istio/wg-user-experience-maintainers review this for me?

sdake reviewed (approved):
LGTM. Refinements can be done as needed. @esnible would you file a bug to track the `ads: default` problem @howardjohn found?

esnible commented:
> LGTM. Refinements can be done as needed. @esnible would you file a bug to track the `ads: default` problem @howardjohn found?

Issue https://github.com/istio/istio/issues/23526 tracks that problem.

istio-testing commented:
@esnible: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | b8a1c42d37ae110c85bf782060a0730f19f0c3da | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/lint_istio/12737) | `/test lint_istio`
integ-local-tests_istio | b8a1c42d37ae110c85bf782060a0730f19f0c3da | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/integ-local-tests_istio/4329) | `/test integ-local-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@esnible: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | b8a1c42d37ae110c85bf782060a0730f19f0c3da | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/lint_istio/12737) | `/test lint_istio`
integ-local-tests_istio | b8a1c42d37ae110c85bf782060a0730f19f0c3da | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/integ-local-tests_istio/4329) | `/test integ-local-tests_istio`
integ-pilot-k8s-tests_istio | b8a1c42d37ae110c85bf782060a0730f19f0c3da | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/integ-pilot-k8s-tests_istio/12798) | `/test integ-pilot-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@esnible: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | b8a1c42d37ae110c85bf782060a0730f19f0c3da | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/lint_istio/12737) | `/test lint_istio`
integ-local-tests_istio | b8a1c42d37ae110c85bf782060a0730f19f0c3da | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/integ-local-tests_istio/4329) | `/test integ-local-tests_istio`
integ-pilot-k8s-tests_istio | b8a1c42d37ae110c85bf782060a0730f19f0c3da | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/integ-pilot-k8s-tests_istio/12798) | `/test integ-pilot-k8s-tests_istio`
integ-multicluster-k8s-tests_istio | b8a1c42d37ae110c85bf782060a0730f19f0c3da | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/integ-multicluster-k8s-tests_istio/3243) | `/test integ-multicluster-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

esnible committed:
Galley no longer present

howardjohn reviewed (changes requested):
looks like a bad merge

istio-testing commented:
@esnible: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-multicluster-k8s-tests_istio | b8a1c42d37ae110c85bf782060a0730f19f0c3da | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/integ-multicluster-k8s-tests_istio/3243) | `/test integ-multicluster-k8s-tests_istio`
integ-pilot-k8s-tests_istio | 84920a88de89c540928417d9e38a90bb33f78d89 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/integ-pilot-k8s-tests_istio/12807) | `/test integ-pilot-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

esnible commented:
@googlebot I fixed it.

esnible commented:
I fixed it.  Question.  I did `git rebase upstream/master`, made some changes, added, committed, pushed.
The PR report hundreds of files chosen.  You commented there was a bad merge.

To get it back to 2 files, I clicked "Edit" near the title, changed to another base, did it again, to change it back to be based on `master`.  That made the PR look normal.  What `git` command should I have used instead?

istio-testing commented:
@esnible: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-pilot-k8s-tests_istio | 84920a88de89c540928417d9e38a90bb33f78d89 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/integ-pilot-k8s-tests_istio/12808) | `/test integ-pilot-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

esnible commented:
/retest

istio-testing commented:
@esnible: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-pilot-k8s-tests_istio | 84920a88de89c540928417d9e38a90bb33f78d89 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/integ-pilot-k8s-tests_istio/12815) | `/test integ-pilot-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

esnible commented:
This needs to go on hold until https://github.com/istio/istio/pull/23724 merges.  A performance improvement in `istioctl proxy-status <pod>` is keeping this test from passing.  When that PR merges it it will be safe to retest this one.

esnible commented:
/retest

istio-testing commented:
@esnible: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-pilot-k8s-tests_istio | 84920a88de89c540928417d9e38a90bb33f78d89 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23502/integ-pilot-k8s-tests_istio/12859) | `/test integ-pilot-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

esnible committed:
Force sort, even on unrecognized typed clusters

esnible commented:
Added a fix that will also resolve https://github.com/istio/istio/issues/23735

esnible commented:
@howardjohn Can you re-review?

istio-testing commented:
In response to a cherrypick label: new pull request created: #23766
